pkg/kvstore/etcd.go

```diff
@@ -81,6 +81,9 @@ func newEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 		}
 	}
 	if config != nil {
+		if config.DialTimeout == 0 {
+			config.DialTimeout = 10 * time.Second
```

Comment:
Failing is good. It will cause the pod to be restarted with an error condition which will indicate to the user that something is not OK. If we retry forever there is no indication to the user that something is going wrong.